pkg/api/types.go

```diff
@@ -398,6 +402,21 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts is the endpoint name that details Glusterfs topology
+	Hosts string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options such as "rw" or "ro"
+	MountOpt string `json:"mountOptions,omitempty"`
```

Comment:
Problem with blacklist is that if glusterfs adds a dangerous option, and a cluster admin upgrades cluster, then there is a security hole.  It puts the burden on Kubernetes project to track everything Gluster project does.  Not the model we are used to.